sympy/utilities/codegen.py

```diff
@@ -177,13 +169,20 @@ def __init__(self, name, arguments, results, local_vars, global_vars):
                 raise ValueError("Unknown Routine result: %s" % r)
             symbols.update(r.expr.free_symbols)
 
+        local_symbols = set([])
+        for r in local_vars:
+            if not isinstance(r, Result):
+                raise ValueError("Unknown Routine result: %s" % r)
```

Comment:
ahh.. thank you, that led me to some use cases of the local_vars before. I was having a challenge understanding what they did (iterate through arrays) and I definitely broke some of that compatibility. I'll work on this.